File: apps/android/app/src/main/kotlin/ai/parkshield/android/feature/auth/domain/AuthModels.kt
package ai.parkshield.android.feature.auth.domain

data class TokenPair(
    val accessToken: String,
    val refreshToken: String,
    val expiresInSeconds: Long,
    val tokenType: String,
)

data class AuthenticatedUser(val id: String, val email: String, val role: String, val isVerified: Boolean)

sealed interface AuthFailure {
    data class Remote(val code: String, val correlationId: String?) : AuthFailure
    data object SessionInvalid : AuthFailure
    data object Transport : AuthFailure
    data object SecureStorage : AuthFailure
}

sealed interface AuthResult<out T> {
    data class Success<T>(val value: T) : AuthResult<T>
    data class Failure(val error: AuthFailure) : AuthResult<Nothing>
}

interface AuthenticationRepository {
    suspend fun login(email: String, password: CharArray): AuthResult<TokenPair>
    suspend fun refresh(refreshToken: String): AuthResult<TokenPair>
    suspend fun logout(refreshToken: String): AuthResult<Unit>
    suspend fun profile(accessToken: String): AuthResult<AuthenticatedUser>
}

File: apps/ios/Sources/ParkShieldFoundation/Feature/Auth/Domain/AuthModels.swift
import Foundation

public struct TokenPair: Codable, Equatable, Sendable {
    let accessToken: String; let refreshToken: String; let expiresIn: Int; let tokenType: String
    enum CodingKeys: String, CodingKey { case accessToken = "access_token", refreshToken = "refresh_token", expiresIn = "expires_in", tokenType = "token_type" }
}

public struct AuthenticatedUser: Codable, Equatable, Sendable {
    let id: String; let email: String; let role: String; let isVerified: Bool
    enum CodingKeys: String, CodingKey { case id, email, role; case isVerified = "is_verified" }
}

public enum AuthFailure: Error, Equatable, Sendable { case remote(code: String, correlationID: String?), sessionInvalid, transport, secureStorage }

public protocol AuthenticationRepository: Sendable {
    func login(email: String, password: String) async -> Result<TokenPair, AuthFailure>
    func refresh(refreshToken: String) async -> Result<TokenPair, AuthFailure>
    func logout(refreshToken: String) async -> Result<Void, AuthFailure>
    func profile(accessToken: String) async -> Result<AuthenticatedUser, AuthFailure>
}
